fix(ngAria): don't attach roles to native controls

m-amr opened the pull request:
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
  fix
- **What is the current behavior?** (You can also link to an open issue here)

attach roles to these native element (textarea, select, button)
- **What is the new behavior (if this is a feature change)?**

don't attach roles to these native these element (textarea, select, button)
- **Does this PR introduce a breaking change?**

no
- **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#commit-message-format
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- **Other information**:

don't add roles to (textarea, button, select, input)
fix #14076

m-amr commented:
I can't found anything to be updated in the documentation.

gkalpak commented on test/ngAria/ariaSpec.js:
```diff
@@ -272,6 +272,37 @@ describe('$aria', function() {
+      compileElement('<a ng-click="doClick()"></a>');
+      expect(element.attr('role')).toBe(undefined);
+    });
+
```
Nitpick: Redundant newline

gkalpak commented on test/ngAria/ariaSpec.js:
```diff
@@ -272,6 +272,37 @@ describe('$aria', function() {
+
+    it('should not add a role to a native select', function() {
+      compileElement('<select type="checkbox" ng-model="val"></select>');
+      expect(element.attr('role')).toBe(undefined);
```
Nitpick: Use `toBeUndefined()` for consistency.

gkalpak commented:
Not super important, but it's more concise to use the `they` helper for tests (since they are pretty repetitive).
Could you also update the commit message to mention this is a BC (albeit a small one that shouldn't affect a11y) ?

m-amr committed:
fix(ngAria): don't add roles to native elements

prevent ngAria from attaching roles to textarea, button, select, summary, details, a and input

Closes  #14076

BREAKING CHANGE:

Curreny behavior is that ngAria don't attach roles to input elements but it does so to other native controls.
so this change will prevent ngAria from attaching roles to native elements like (textrea, select, button, etc) and it will only attach roles to non-native controls.
and this should not affect accessibility, because native inputs are accessible by default.
(albeit a small one that shouldn't affect a11y)

m-amr commented:
Thanks fot your feedback, I have update the code
1 - I am now using 'they' instead of 'it' 
2 - I have updated the commit message.

Thanks.

Narretz commented:
The commit message must still be updated to conform with our standard on Breaking changes: https://docs.google.com/document/d/1QrDFcIiPjSLDn3EL15IJygNPiHORgU1_OOAqWjiDU5Y/edit#heading=h.gbbngquhe0qa

m-amr commented:
@Narretz  
I have update the commit message to confirm the standard of the breaking change.
please check it.
Thanks

Narretz commented:
@m-amr you still need to add an actual explanation of the breaking change, because that will show up in the changelog, e.g. roles are not added to native input / form elements anymore, and that this should not affect accessibility, because native inputs are accessible by default.

m-amr commented:
@Narretz i have added some explanation about the change
could you please check it again ?
